ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano/Orphans.hs

```diff
@@ -0,0 +1,187 @@
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE DerivingVia                #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase                 #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeOperators              #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+{-# LANGUAGE NamedFieldPuns             #-}
+module Ouroboros.Consensus.Cardano.Orphans () where
+
+import           Cardano.Binary
+import           Cardano.Chain.Common (Address, BlockCount, CompactAddress,
+                     KeyHash, Lovelace)
+import           Cardano.Chain.Delegation (Certificate)
+import           Cardano.Chain.Genesis
+import           Cardano.Chain.UTxO
+import           Cardano.Chain.Update (ProtocolParameters)
+import           Cardano.Crypto.ProtocolMagic
+import           Cardano.Crypto.Signing.Redeem
+import           Data.Map (Map)
+import           Data.Set (Set)
+import           Data.Word (Word64)
+import           Ouroboros.Consensus.Util.Orphans ()
+
+{-------------------------------------------------------------------------------
+  Orphan Instances
```

Comment:
These all look reasonable, and I like the type applications for avoiding silent changes.

I do wonder: maybe we should consult with the owners of these data types before (potentially) committing to a serialization. Just in case they have plans to eg change these types.

@nc6 @JaredCorduan --- What do you think? Are these types stable and these serializations appropriate? Thanks.

Reply:
I made a WIP PR in the mean time. https://github.com/input-output-hk/cardano-ledger-specs/pull/2202

Reply:
I see that it's merged upstream; great!